src/readmeoss/agent/readmeoss.php

```diff
@@ -141,11 +141,38 @@ function processUploadId($uploadId)
     }
 
     $contents = array('licensesMain'=>$licenseStmtsMain, 'licenses'=>$licenseStmts, 'copyrights'=>$copyrightStmts, 'licenseAcknowledgements' => $licenseAcknowledgements);
+    $contents = $this->identifiedGlobalLicenses($contents);
```

Comment:
This line is not very readable and it mutates `$contents`, maybe add a new variable for the result with a descriptive name.